File: leetcode/find-words-that-can-be-formed-by-characters.py
class Solution:
    def countCharacters(self, words: list[str], chars: str) -> int:
        dw = {}
        fo = 0
        for char in chars:
            if char in dw:
                dw[char] += 1
            else:
                dw[char] = 1
        for x in words:
            brek = False
            d = {}
            for char in x:
                if char in d:
                    d[char] += 1
                else:
                    d[char] = 1
            for char in d:
                if char in dw:
                    pass
                else:
                    brek = True
                    break
            if brek != True:
                for char in d:
                    if d[char] <= dw[char]:
                        pass
                    else:
                        brek = True
                        break
            if brek != True:
                fo += len(x)
        return fo

File: leetcode/find-words-that-can-be-formed-by-characters.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
int countCharacters(vector<string>& words, string chars) {
    int fo = 0;
    unordered_map<char, int> dw;
    unordered_map<char, int> d;
    bool cond;

    for (char c : chars) {
        if (dw.find(c) == dw.end()) {
            dw[c] = 1;
        }
        else {
            dw[c]++;
        }
    }

    for (string x : words) {
        d.clear();
        cond = false;

        for (char c : x) {
            if (d.find(c) == d.end()) {
                d[c] = 1;
            }
            else {
                d[c]++;
            }
        }

        for (const auto& pair : d) {
            if (dw.find(pair.first) == dw.end()) {
                cond = true;
                break;
            }
        }
        if (!cond) {
            for (const auto& pair : d) {
                if (dw[pair.first] < d[pair.first]) {
                    cond = true;
                    break;
                }
            }
        }
        if (!cond) {
            fo += x.size();
        }
    }



return fo;
}
};
